src/index.ts

```diff
@@ -238,8 +235,8 @@ function getByPath<T extends SearchableObject, P extends PathEntry<T> & string>(
   return pathArray.reduce((current: unknown, pathPart) => {
     if (
       typeof current !== 'object' ||
-      current === null ||
-      !hasOwnProperty.call(current, pathPart)
+      // eslint-disable-next-line no-prototype-builtins
+      !current?.hasOwnProperty(pathPart) // We use the builtin to avoid reactivity issues with Proxy
```

Comment:
```suggestion
     current === null ||
      current[pathPart] === undefined || // Trigger dependency tracking for libraries (e.g. vue)
      !hasOwnProperty.call(current, pathPart)
```

I would suggest this. We still use the global `hasOwnProperty` function but signal vue, that we access the property so its depedency tracking works.